data_fetcher.py

```diff
@@ -0,0 +1,45 @@
+import bs4
+import requests
+import lxml
+from bs4 import BeautifulSoup
+from typing import List
+ 
+ 
+class JeuneAfriqueRSSDataSource:
+    
+    def __init__(self, rss_url: str) -> None:
+        self.url = rss_url
+        self.r = None
+        self.status_code = None
+        self.soup = None
+        self.articles = None
+        self.articles_dicts = None
+        self.urls: List[str] = []
+        self.titles: List[str] = []
+        self.descriptions: List[str] = []
+        self.pub_dates: List[str] = []
+        
+    def get_data(self) -> None:
```

Comment:
When a function has "get" in its name I automatically assume it should return something. In your case, it is mutating a state of  an object `JeuneAfriqueRSSDataSource`, so I would either expect it either:
* to have name "set_something"
* or to return some data, such as a dataclass with all the fields you set here